libs/dh/shared/ui-paginator/src/lib/dh-shared-ui-paginator.component.ts

```diff
@@ -0,0 +1,99 @@
+/**
+ * @license
+ * Copyright 2020 Energinet DataHub A/S
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License2");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+import { Component, OnDestroy, inject, ViewChild, Input } from '@angular/core';
+import {
+  MatPaginator,
+  MatPaginatorIntl,
+  MatPaginatorModule,
+} from '@angular/material/paginator';
+import { TranslocoService } from '@ngneat/transloco';
+import { Subject, takeUntil } from 'rxjs';
+
+@Component({
+  standalone: true,
+  selector: 'dh-shared-ui-paginator',
+  imports: [MatPaginatorModule],
+  template: `
+    <mat-paginator
+      [length]="length"
+      [pageSize]="pageSize"
+      [pageSizeOptions]="pageSizeOptions"
+      [showFirstLastButtons]="true"
+      [attr.aria-label]="ariaLabel"
+    >
+    </mat-paginator>
+  `,
+})
+export class DhSharedUiPaginatorComponent implements OnDestroy {
+  @ViewChild(MatPaginator, { static: true }) private paginator!: MatPaginator;
+
+  @Input() length?: number;
+  @Input() pageSizeOptions: number[] = [50, 100, 150, 200, 250];
+  @Input() pageSize?: number;
+
+  ariaLabel?: string;
+
+  private matPaginatorIntl = inject(MatPaginatorIntl);
+  private translocoService = inject(TranslocoService);
```

Comment:
Noob question: Is there any difference between this and constructor DI?

Reply:
There shouldn't be any differences regarding the dependency injection itself. However, this syntax is not as "magic" as the constructor-based injection and actually, this also makes it possible to make reusable functions using dependency injections. Example:
![image](https://user-images.githubusercontent.com/86852536/196870720-86b703e1-ebc5-44d7-9835-211580ec955e.png)

Reply:
I love it, will definitely use that more!